### PR TITLE
Dịch "minibatch-sgd_vn" - Phần 7

### LINKED ISSUE #1885: Dịch "minibatch-sgd_vn" - Phần 7
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_optimization/minibatch-sgd_vn.md#L510-L539

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by davidnvq**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Điền số issue sau dấu `#` (không có dấu cách): Close #1885

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn. Bạn đã kiểm tra rằng mỗi đoạn văn đã nằm một dòng?

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] Điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Lưu ý cho người phản biện (Reviewer)

1. Chỉ review nếu Pull Request (PR) này đã sẵn sàng cho việc review. Một PR sẵn sàng khi PR đó không có label `status: wip`.
2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

### CODE REVIEW

File: chapter_optimization/minibatch-sgd_vn.md [hunk 1]
```diff
@@ -521,7 +521,12 @@ train_gluon_ch11('sgd', {'learning_rate': 0.05}, data_iter)
 * In general, minibatch SGD is faster than SGD and gradient descent for convergence to a smaller risk, when measured in terms of clock time.  
 -->
 
-*dịch đoạn phía trên*
+* Vector hoá tính toán sẽ giúp mã nguồn hiệu quả hơn do giảm chi phí phát sinh từ framework học sâu và tận dụng tính cục bộ của bộ nhớ và vùng nhớ đệm trên CPU và GPU tốt hơn.
+* Tồn tại sự cân nhắc lựa chọn giữa hiệu quả thống kê từ SGD và hiệu quả tính toán từ việc xử lý các batch dữ liệu với kích thước lớn tại một thời điểm.
```
Comment (cuongvng) [inline chapter_optimization/minibatch-sgd_vn.md:525]:
```suggestion
* Tồn tại sự đánh đổi giữa hiệu quả thống kê của SGD và hiệu quả tính toán từ việc xử lý các batch dữ liệu kích thước lớn tại một thời điểm.
```

File: chapter_optimization/minibatch-sgd_vn.md [hunk 2]
```diff
@@ -521,7 +521,12 @@ train_gluon_ch11('sgd', {'learning_rate': 0.05}, data_iter)
...
+* Thuật toán hạ gradient ngẫu nhiên theo minibatch mang đến điều tốt nhất của cả hai lựa chọn trên: hiệu quả tính toán và thống kê.
+* Trong thuật toán SGD theo minibatch chúng ta xử lý các batch dữ liệu thu được từ hoán vị ngẫu nhiên của dữ liệu huấn luyện (cụ thể, mỗi quan sát được xử lý chỉ một lần mỗi epoch theo thứ tự ngẫu nhiên).
+* Suy giảm tốc độ trong quá trình huấn luyện được khuyến khích sử dụng.
+* Nói chung, thuật toán SGD theo minibatch nhanh hơn so với thuật toán SGD và hạ descent về tốc độ hội tụ tới điểm giá trị mất mát nhỏ hơn khi đo thời gian đồng hồ (_clock time_).
```
Comment (cuongvng) [inline chapter_optimization/minibatch-sgd_vn.md:529]:
```suggestion
* Nói chung, thuật toán SGD theo minibatch nhanh hơn thuật toán SGD và GD về thời gian hội tụ.
```
chắc thế này cũng đủ ý.

File: chapter_optimization/minibatch-sgd_vn.md [hunk 3]
```diff
@@ -534,7 +539,11 @@ train_gluon_ch11('sgd', {'learning_rate': 0.05}, data_iter)
 How does the behavior of SGD, minibatch SGD and that of gradient descent change?
 -->
 
-*dịch đoạn phía trên*
+1. Sửa đổi kích thước batch và tốc độ học và quan sát tốc độ suy giảm giá trị của hàm mục tiêu và thời gian cần thiết trong mỗi epoch.
```
Comment (cuongvng) [inline chapter_optimization/minibatch-sgd_vn.md:542]:
```suggestion
1. Sửa đổi kích thước batch và tốc độ học, quan sát tốc độ suy giảm giá trị của hàm mục tiêu và thời gian cho mỗi epoch.
```

File: chapter_optimization/minibatch-sgd_vn.md [hunk 4]
```diff
@@ -534,7 +539,11 @@ train_gluon_ch11('sgd', {'learning_rate': 0.05}, data_iter)
...
+2. Đọc thêm tài liệu MXNet và sử dụng lớp `Trainer`  với hàm `set_learning_rate` để giảm tốc độ học của SGD theo minibatch bằng 1/10 giá trị trước đó sau mỗi epoch.
```
Comment (cuongvng) [inline chapter_optimization/minibatch-sgd_vn.md:543]:
```suggestion
2. Đọc thêm tài liệu MXNet và sử dụng lớp `Trainer`  với hàm `set_learning_rate` để giảm tốc độ học của SGD theo minibatch còn 1/10 giá trị trước đó sau mỗi epoch.
```

File: chapter_optimization/minibatch-sgd_vn.md [hunk 5]
```diff
@@ -534,7 +539,11 @@ train_gluon_ch11('sgd', {'learning_rate': 0.05}, data_iter)
...
+3. Hãy so sánh SGD theo minibatch với một biến thể mà *chọn mẫu có hoàn lại* từ tập huấn luyện. Điều gì sẽ xảy ra?
```
Comment (cuongvng) [inline chapter_optimization/minibatch-sgd_vn.md:544]:
```suggestion
3. Hãy so sánh SGD theo minibatch với một biến thể *lấy mẫu có hoàn lại* từ tập huấn luyện. Điều gì sẽ xảy ra?
```

File: chapter_optimization/minibatch-sgd_vn.md [hunk 6]
```diff
@@ -534,7 +539,11 @@ train_gluon_ch11('sgd', {'learning_rate': 0.05}, data_iter)
...
+4. Một ác thần sao chép bộ dữ liệu của bạn mà không nói cho bạn biết (cụ thể, mỗi quan sát xảy ra hai lần và dữ liệu của bạn bằng hai lần kích thước ban đầu của nó, nhưng không ai nói với bạn biết).
+Cách hoạt động của thuật toán SGD, SGD theo minibatch và thuật toán hạ gradient sẽ thay đổi như thế nào?
```
Comment (cuongvng) [inline chapter_optimization/minibatch-sgd_vn.md:546]:
```suggestion
Cách hoạt động của các thuật toán hạ gradient, SGD và SGD theo minibatch sẽ thay đổi như thế nào?
```

File: chapter_optimization/minibatch-sgd_vn.md [hunk 7]
```diff
@@ -534,7 +539,11 @@ train_gluon_ch11('sgd', {'learning_rate': 0.05}, data_iter)
 How does the behavior of SGD, minibatch SGD and that of gradient descent change?
 -->
 
-*dịch đoạn phía trên*
+1. Sửa đổi kích thước batch và tốc độ học, quan sát tốc độ suy giảm giá trị của hàm mục tiêu và thời gian cho mỗi epoch.
+2. Đọc thêm tài liệu MXNet và sử dụng lớp `Trainer`  với hàm `set_learning_rate` để giảm tốc độ học của SGD theo minibatch bằng 1/10 giá trị trước đó sau mỗi epoch.
+3. Hãy so sánh SGD theo minibatch với một biến thể *lấy mẫu có hoàn lại* từ tập huấn luyện. Điều gì sẽ xảy ra?
+4. Một ác thần sao chép bộ dữ liệu của bạn mà không nói cho bạn biết (cụ thể, mỗi quan sát xảy ra hai lần và dữ liệu của bạn bằng hai lần kích thước ban đầu của nó, nhưng không ai nói với bạn biết).
```
Comment (minhduc0711) [inline chapter_optimization/minibatch-sgd_vn.md:545]:
```suggestion
4. Một ác thần đã sao chép tập dữ liệu của bạn mà không nói cho bạn biết (cụ thể, mỗi quan sát bị lặp lại hai lần và kích thước tập dữ liệu tăng gấp đôi so với ban đầu, nhưng không ai nói với bạn biết).
```

File: chapter_optimization/minibatch-sgd_vn.md [hunk 8]
```diff
@@ -521,7 +521,12 @@ train_gluon_ch11('sgd', {'learning_rate': 0.05}, data_iter)
 * In general, minibatch SGD is faster than SGD and gradient descent for convergence to a smaller risk, when measured in terms of clock time.  
 -->
 
-*dịch đoạn phía trên*
+* Vector hoá tính toán sẽ giúp mã nguồn hiệu quả hơn do giảm chi phí phát sinh từ framework học sâu và tận dụng tính cục bộ của bộ nhớ và vùng nhớ đệm trên CPU và GPU tốt hơn.
+* Tồn tại sự đánh đổi giữa hiệu quả thống kê của SGD và hiệu quả tính toán từ việc xử lý các batch dữ liệu kích thước lớn tại một thời điểm.
+* Thuật toán hạ gradient ngẫu nhiên theo minibatch mang đến điều tốt nhất của cả hai lựa chọn trên: hiệu quả tính toán và thống kê.
+* Trong thuật toán SGD theo minibatch chúng ta xử lý các batch dữ liệu thu được từ hoán vị ngẫu nhiên của dữ liệu huấn luyện (cụ thể, mỗi quan sát được xử lý chỉ một lần mỗi epoch theo thứ tự ngẫu nhiên).
+* Suy giảm tốc độ trong quá trình huấn luyện được khuyến khích sử dụng.
+* Nói chung, thuật toán SGD theo minibatch nhanh hơn thuật toán SGD và GD về thời gian hội tụ.
```
[the remark this answers is not in the document]
Comment (davidnvq) [reply chapter_optimization/minibatch-sgd_vn.md:529]: risk trong ML thường là loss. Tuy nhiên đoạn `tốt hơn` thì không hẳn, 1 lần cập nhất của GD bao giờ cũng có độ `generalization` tốt nhất.

[the remark this answers is not in the document]
Comment (minhduc0711) [reply chapter_optimization/minibatch-sgd_vn.md:529]: ừm, mình vừa đọc lại thì hiểu ý tác giả là "hội tụ nhanh hơn xuống cùng 1 giá trị mất mát"
